foreshadow/newintents/intents.py

```diff
@@ -0,0 +1,158 @@
+"""Base package for all intent definitions."""
+
+from functools import partial
+
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from foreshadow.metrics.internals import (
+    is_numeric,
+    is_string,
+    num_valid,
+    unique_heur,
+)
+
+
+class BaseIntent(BaseEstimator, TransformerMixin):
+    """Base for all intent definitions.
+
+    For each intent subclass a class attribute called `confidence_computation`
+    must be defined which is of the form::
+
+       {
+            metric_def: weight
+       }
+
+    """
+
+    @classmethod
+    def get_confidence(cls, X, y=None):
+        """Determine the confidence for an intent match.
+
+        Args:
+            X: input DataFrame.
+            y: response variable
+
+        Returns:
+            float: A confidence value bounded between 0.0 and 1.0
+
+        """
+        return sum(
+            [
+                metric_fn(X) * weight
+                for metric_fn, weight in cls.confidence_computation.items()
+            ]
+        )
+
+
+class Numeric(BaseIntent):
+    """Defines a numeric column type."""
+
+    confidence_computation = {
+        num_valid: 0.25,
+        partial(unique_heur, invert=True): 0.25,
```

Comment:
Let's discuss this. I'm still debating in my head how we want the invert to work. This looks clean and honestly a user shouldn't be using a metric if they don't understand which direction is better. I just wonder if we ever see these being used in loops where passing invert = True to some items and passing invert = False to others won't be convenient. Best thing might be to create an issue for this and see if we hit this use case.

Reply:
Sure, lets chat

Reply:
Another method is to use True/False second output which answers the question "is higher better?". Then if you have conflicting . bools, you swap them all so that we can compare. Let's leave it this way for now as this appears cleaner.